### PR TITLE
✨ Added products listing page

### PR DESCRIPTION
**Pull request by shaksham08**
Added products listing page with filters

- Users can see list of products and also filter them as per the filters provided

### CODE REVIEW

File: src/context/ProductsContext/ProductContext.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  categories: [],
+  selectedCategories: {},
+};
+
+const ProductContext = createContext();
+
+const ProductContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(ProductReducer, initialState);
+
+  useEffect(() => {
```
Comment (anshikaa12) [inline src/context/ProductsContext/ProductContext.js:22]: you can make a services folder and extract this API fetching part there!

**Review by anshikaa12 — APPROVED** [review]

File: src/context/ProductsContext/ProductContext.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      dispatch({
+        type: ProductActions.GET_DATA,
+        payload: {
+          products,
+          categories,
+        },
+      });
+    })();
+  }, []);
```
Comment (Raghav888) [inline src/context/ProductsContext/ProductContext.js:45]: You can add this code in utils folder

File: src/context/ProductsContext/ProductReducer.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    case ProductActions.FILTER_RATING:
+      return { ...state, ratings: action.payload.value };
+
+    case ProductActions.FILTER_CATEGORY:
+      let updatedCategories = { ...state.selectedCategories };
+      updatedCategories[action.payload.key] = action.payload.value;
+      return { ...state, selectedCategories: updatedCategories };
+
+    case ProductActions.RESET_FILTER:
```
Comment (Raghav888) [inline src/context/ProductsContext/ProductReducer.js:31]: I am doubtful, whether to write case in this way is good practice or not

**Review by Raghav888 — APPROVED** [review]
1. Like the structure of code
2. Good naming convention
3.  Try to raise small PR

File: src/context/ProductsContext/ProductReducer.js (hunk 3 above)
Comment (shaksham08) [reply src/context/ProductsContext/ProductReducer.js:31]: Yes its one of the way we can write action types.

**Pull request merged**